src/main/java/com/team766/orin/GetApriltagPoseData.java

```diff
@@ -0,0 +1,41 @@
+package com.team766.orin;
+
+import edu.wpi.first.apriltag.AprilTag;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Translation3d;
+import java.util.ArrayList;
+
+public class GetApriltagPoseData {
+
+    public static ArrayList<AprilTag> getAllTags() {
+        ArrayList<AprilTag> apriltags = new ArrayList<AprilTag>();
+
+        int[] tagIds;
+        double[] tagData;
+
+        try {
+            tagIds = GetOrinRawValue.getTagIds();
+            tagData = GetOrinRawValue.getRawPoseData();
```

Comment:
there's a [race condition](https://stackoverflow.com/questions/34510/what-is-a-race-condition) here where the TagIds and RawPoseData entries aren't updated at the same time. for example, you could get old TagIds with new RawPoseData, or vice versa.

Reply:
I see (and agree). How do you recommend preventing against this? I wish there was a getTimeStamp function or something, and we could compare the two timestamps.

In reality, would the best idea to be to ensure that there is the correct number of data points per tag (Line #26)? Or else do some on-field testing and see what happens? Thoughts?

Reply:
I would store everything (both tag IDs and poses) in a single array of `double`s. A `double` gives you [53 bits of precision](https://blogs.oracle.com/javamagazine/post/java-primitive-datatypes-int-float-double), so it can perfectly store every value that a 32-bit `int` can.

Reply:
Ah! That makes a lot of sense! Updating it on the C++ side and next the Java side!

Reply:
Fixed!